flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1249,6 +1271,23 @@ def test_attach_unknown_volume(self):
             attach_to=self.this_node,
         )
 
+    def test_attach_volume_validate_size(self):
+        """
+        An attempt to attach an already attached ``BlockDeviceVolume`` raises
+        ``AlreadyAttachedVolume``.
+        """
+        dataset_id = uuid4()
+
+        new_volume = self.api.create_volume(
+            dataset_id=dataset_id,
+            size=REALISTIC_BLOCKDEVICE_SIZE
+        )
+        attached_volume = self.api.attach_volume(
+            new_volume.blockdevice_id, attach_to=self.this_node,
+        )
+
+        self._assert_volume_size(attached_volume, REALISTIC_BLOCKDEVICE_SIZE)
```

Comment:
Thanks for pointing this out! 
There are 2 kinds of comparisons i wanted to do in this test:
1) Device size reported by `lsblk` matches volume size reported in BlockDeviceVolume.
2) Device size reported by `lsblk` matches input volume size at create_volume time.

(1) was implemented and the test passed. i missed doing (2). Added that part, along with FLOC-1874 disclaimer. Thanks for catching!